app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -139,6 +147,41 @@ open class ProfilesViewModel(private val repository: ProfilesRepository) : ViewM
     _loading.value = true
     repository.deleteProfileByUid(
         uid, onSuccess = { _loading.value = false }, onFailure = { e -> handleError(e) })
+    ppRepository.deleteProfilePicture(
+        userId = uid, onSuccess = {}, onFailure = { e -> handleError(e) })
+  }
+
+  /**
+   * Uploads the current user's profile picture to the remote storage system.
+   *
+   * @param imageData The byte array of the image file to be uploaded.
+   * @throws IllegalStateException if the user is not logged in.
+   */
+  fun uploadCurrentUserProfilePicture(imageData: ByteArray) {
+    val userId = selectedProfile.value?.uid ?: throw IllegalStateException("User not logged in")
+    ppRepository.uploadProfilePicture(
+        userId = userId,
+        imageData = imageData,
+        onSuccess = { url ->
+          _selectedProfile.update { selected -> selected?.copy(profilePictureUrl = url) }
```

Comment:
_selectedProfile can be the profile of someone else. You need to make sure that this function is only called when _selectedProfile is set to your profile. If it is the case it seems good for me. Just check for that :)

Reply:
in theory, when you select your profile for edition, _selectedProfile is updated to your profile and stays there until you select someone else's profile. So as the profile picture upload option is only available from the personal edit screen, there is no chance to edit someone else's profile.

In practice, It is not impossible that a bug make another user's profile appear in the personal edit screen and create the vulnerability you are concerned about.

So I'll make a double security check , but in the next PR where I'll implement image upload from edit screen.